src/bin/cmd/server.rs

```diff
@@ -45,21 +45,16 @@ fn start_server_tui(config: servers::ServerConfig) {
 	// everything it might need
 	if config.run_tui.unwrap_or(false) {
 		warn!("Starting GRIN in UI mode...");
-		servers::Server::start(config, |serv: Arc<servers::Server>| {
-			let running = Arc::new(AtomicBool::new(true));
-			let _ = thread::Builder::new()
-				.name("ui".to_string())
-				.spawn(move || {
```

Comment:
So we now run the TUI in the main thread and not in a separate ui thread?
Is there any downside to this?

Reply:
It's symmetrical with non-tui case. Also  to be fair I'm not sure, it seems that `cursive` creates a separate thread anyway, but I'm not an expert. What I see is that SIGINT is magically handled (in ncurses backend or most likely in C code, because I didn't find any traces of it in `cursive`) and it stops only UI, so we are able to call `server.stop()` in the main thread after that.

Reply:
:+1: